static/app/views/settings/projectDebugFiles/index.tsx

```diff
@@ -27,211 +34,168 @@ type Props = RouteComponentProps<{projectId: string}, {}> & {
-    });
-
-    this.api.request(
-      `/projects/${organization.slug}/${params.projectId}/files/dsyms/?id=${id}`,
+  const {
+    data: debugFiles,
+    getResponseHeader: getDebugFilesResponseHeader,
+    isLoading: isLoadingDebugFiles,
+    refetch: refetchDebugFiles,
+  } = useApiQuery<DebugFile[] | null>(
+    [
+      `/projects/${organization.slug}/${params.projectId}/files/dsyms/`,
       {
-        method: 'DELETE',
-        complete: () => this.fetchData(),
-      }
-    );
-  };
-
-  handleSearch = (query: string) => {
-    const {location, router} = this.props;
-
-    router.push({
-      ...location,
-      query: {...location.query, cursor: undefined, query},
-    });
-  };
-
-  async fetchProject() {
-    const {organization, params} = this.props;
-    try {
-      const updatedProject = await this.api.requestPromise(
-        `/projects/${organization.slug}/${params.projectId}/`
-      );
-      ProjectsStore.onUpdateSuccess(updatedProject);
-    } catch {
-      addErrorMessage(t('An error occurred while fetching project data'));
+        query: {query: location.query.query},
+      },
+    ],
+    {
+      staleTime: 0,
     }
-  }
-
-  getQuery() {
-    const {query} = this.props.location.query;
-
-    return typeof query === 'string' ? query : undefined;
-  }
-
-  getEmptyMessage() {
-    if (this.getQuery()) {
-      return t('There are no debug symbols that match your search.');
+  );
+
+  const {
+    data: builtinSymbolSources,
+    isLoading: isLoadingSymbolSources,
+    refetch: refetchBuiltinSymbolSources,
+  } = useApiQuery<BuiltinSymbolSource[] | null>(
+    [`/organizations/${organization.slug}/builtin-symbol-sources/`],
+    {
+      staleTime: 0,
+      enabled: organization.features.includes('symbol-sources'),
     }
-
-    return t('There are no debug symbols for this project.');
-  }
-
-  renderLoading() {
-    return this.renderBody();
-  }
-
-  renderDebugFiles() {
-    const {debugFiles, showDetails} = this.state;
-    const {organization, params, project} = this.props;
-
-    if (!debugFiles?.length) {
-      return null;
-    }
-
-    return debugFiles.map(debugFile => {
-      const downloadUrl = `${this.api.baseUrl}/projects/${organization.slug}/${params.projectId}/files/dsyms/?id=${debugFile.id}`;
-
-      return (
-        <DebugFileRow
-          debugFile={debugFile}
-          showDetails={showDetails}
-          downloadUrl={downloadUrl}
-          downloadRole={organization.debugFilesRole}
-          onDelete={this.handleDelete}
-          key={debugFile.id}
-          orgSlug={organization.slug}
-          project={project}
-        />
+  );
+
+  const handleSearch = useCallback(
+    (query: string) => {
+      navigate({
+        ...location,
+        query: {...location.query, cursor: undefined, query},
+      });
+    },
+    [navigate, location]
+  );
+
+  const query = useMemo(() => {
+    return typeof location.query === 'string' ? location.query : undefined;
+  }, [location.query]);
```

Comment:
As the return value of this `useMemo` is atomic (`string | undefined`) it will actually not prevent any re-renders, right?
Also the original code was accessing `location.query.query`
So we could just do:
```ts
const queryParam = location.query.query
const query = queryParam === 'string' ? queryParam : undefined
```

Reply:
good catch!!